common/djangoapps/third_party_auth/pipeline.py

```diff
@@ -885,3 +878,40 @@ def get_username(strategy, details, backend, user=None, *args, **kwargs):
     else:
         final_username = storage.user.get_username(user)
     return {'username': final_username}
+
+
+def set_learner_active_enterprise(user=None, backend=None, *args, **kwargs):
+    """
+    Make 'active' a user's enterprise,
+    if the currently 'active' enterprise in EnterpriseCustomerUser does not match the SAML idp-enterprise
+    """
+    from openedx.features.enterprise_support.api import (
+        EnterpriseApiClient,
+        EnterpriseApiException,
+        get_enterprise_learner_data,
+    )
+    if is_provider_saml(backend.name, kwargs):
+        saml_prefix = 'saml-'
+        idp_name = kwargs['response']['idp_name']
+        idp_name = saml_prefix + idp_name
+        enterprise_learner_data = get_enterprise_learner_data(user)
+        enterprise_learner_data_idp = enterprise_learner_data[0]['enterprise_customer'][
+            'identity_provider'] if enterprise_learner_data else None
+
+        if enterprise_learner_data and idp_name != enterprise_learner_data_idp:
+            enterprise_learner = [enterprise_learner for enterprise_learner in enterprise_learner_data if
+                                  enterprise_learner['enterprise_customer']['identity_provider'] == idp_name]
+            if enterprise_learner:
+                uuid = enterprise_learner[0]['enterprise_customer']['uuid']
+                try:
+                    EnterpriseApiClient(user=user).post_active_enterprise_customer(user.username, uuid, True)
```

Comment:
we should create a new method `activate_learner_enterprise` in `enterprise_support\api` and that method should be responsible for activating a given enterprise for a learner and it should update newly activated enterprise in learner's session.